### PR TITLE
define casper interface to avoid cyclic import

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: core/blockchain.go [hunk 1]
```diff
@@ -127,7 +129,8 @@ type BlockChain struct {
 	validator Validator // block and state validator interface
 	vmConfig  vm.Config
 
-	badBlocks *lru.Cache // Bad block cache
+	badBlocks *lru.Cache                                           // Bad block cache
```
Comment (kimmylin) [inline core/blockchain.go]: don't change this one

File: core/blockchain.go [hunk 2]
```diff
@@ -958,7 +961,7 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 	}
 	// If the total difficulty is higher than our known, add it to the canonical chain
 	setNewHead := externTd.Cmp(localTd) > 0
-	if bc.chainConfig.IsCasper(block.Number()) {
+	if bc.casperGen != nil && bc.chainConfig.IsCasper(block.Number()) {
```
Comment (kimmylin) [inline core/blockchain.go]: could wrap this check into a function inside casper_util.go. but up to you.

File: core/blockchain.go (hunk 1 above)
Comment (markya0616) [reply core/blockchain.go]: It's changed by `gofmt`.

File: core/blockchain.go (hunk 2 above)
Comment (markya0616) [reply core/blockchain.go]: OK

**Pull request merged**